assets/js/components/Dashboard/PaMessagesPage.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { ComponentType, useState, useEffect } from "react";
+import { Container, Row, Col, Button, FormCheck } from "react-bootstrap";
+import { PlusCircleFill } from "react-bootstrap-icons";
+import { fetchPaMessages } from "../../utils/api";
+import { PaMessage } from "../../models/pa_message";
+
+const PaMessagesPage: ComponentType = () => {
+  const [paMessages, setPaMessages] = useState<PaMessage[]>([]);
+
+  useEffect(() => {
+    fetchPaMessages().then((allPaMessages) => {
+      setPaMessages(allPaMessages);
+    });
```

Comment:
**note (non-blocking):** I know that React isn't thrilled when you return a Promise (and thus also when you use an `async` function) from a `useEffect` callback. If you'd like to stick with `async`/`await` (I know I like to...) you could create a new `async` function within the callback and call it.

<details>
<summary>Suggested change 📝 </summary>

```suggestion
    const fetchAndSetPaMessages = async () => {
      const allPaMessages = await fetchPaMessages()
      setPaMessages(allPaMessages)
    }
    fetchAndSetPaMessages();
```
</details>

**issue (non-blocking):** Additionally, it looks like this page does not handle loading or errors states for fetching PA messages. If loading PA messages takes a while to load either due to server response time or just a slower connection _or_ the request fails altogether the page will erroneously report **"There are no active PA/ESS Messages."**

Is this something we typically handle? Something like [SWR](https://swr.vercel.app/) could do a lot of the additional state tracking required to support handling something like that.

Reply:
1. Works for me!

2. I don't think we typically handle error/loading states in Screenplay with much special treatment, generally speaking, and normally just default to some empty state for sake of simplicity. That said, it could be something worth discussing especially as the application grows along with its user base. Since it wasn't explicitly called out in this task/designs and I could see it requiring some level of design work if we incorporated that here though, I think it could be a little out of scope for this one.